packages/interactivity-router/src/index.ts

```diff
@@ -170,13 +170,15 @@ window.addEventListener( 'popstate', async () => {
 if ( globalThis.IS_GUTENBERG_PLUGIN ) {
 	if ( navigationMode === 'fullPage' ) {
 		// Cache the scripts. Has to be called before fetching the assets.
-		[].map.call( document.querySelectorAll( 'script[src]' ), ( script ) => {
-			headElements.set( script.getAttribute( 'src' ), {
-				tag: script,
-				text: script.textContent,
-			} );
-		} );
-		await fetchHeadAssets( document, headElements );
+		[].map.call(
+			document.querySelectorAll( 'script[type="module"][src]' ),
```

Comment:
That's intriguing, as it skips all scripts that aren't ES Modules moving forward. It never covered inline scripts, to there is no change regarding that.

What's the story for the plugins or themes that add custom JavaScript to the site? It will load on the full-page reload but not when doing the full-page client-side navigation. It would be good to start collecting the requirements for the feature to work correctly.

Reply:
Yes, that's intentional. This was outlined by @luisherranz in https://github.com/WordPress/gutenberg/pull/59707#issuecomment-2106176106: 

> @michalczaplinski We shouldn't really load unknown scripts. We just did it to get it working with the WP movies demo.
>
> We should only load the module scripts that belong to the new blocks on the page or other module scripts that have declared their compatibility with full-page client-side navigation.

One issue with scripts was that they could depend on the [loading order](https://github.com/WordPress/gutenberg/pull/59707#pullrequestreview-1975274682). That's because scripts (without `async` or `deferred` attributes) are blocking. Modules, on the other hand, are deferred by default and users should be able to load them in any order.

Reply:
Let's make sure it's documented in the code so it's clear that it's intentional to support only ES modules served from files.

Reply:
Gotcha. Done in 375d4eea85cbdc382206c6b12c401ed22f6811a7